mappings/mapper.go

```diff
@@ -90,10 +93,23 @@ func (r *RoleMapper) checkRoleForNamespace(roleArn string, namespace string) boo
 	ar := kube2iam.GetNamespaceRoleAnnotation(ns, r.namespaceKey)
 	for _, rolePattern := range ar {
 		normalized := r.iam.RoleARN(rolePattern)
+
+		if strings.ToLower(r.namespaceRestrictionFormat) == "regexp" {
+			matched, err := regexp.MatchString(normalized, roleArn)
+			if err != nil {
+				log.Errorf("Namespace annotation %s caused an error when trying to match: %s for namespace: %s", rolePattern, roleArn, namespace)
+				return false
+			}
+			if matched {
+				log.Debugf("Role: %s matched %s on namespace:%s.", roleArn, rolePattern, namespace)
+			}
+			return matched
```

Comment:
Haven't had a chance to try this out, but it seems this would "fast fail" on a set of regexps to process and the first one doesn't match.  

We should add a test case (for both Glob and Regexp) that proves out that it will attempt to match against *every* role not just the first.

Reply:
@jrnt30  yes you're right, we only ever have one annotation so I didn't think about fast fail behaviour. I'll look at improving this and adding some tests for multiple annotations